include/drivers/dma.h

```diff
@@ -327,6 +364,78 @@ static inline int z_impl_dma_stop(const struct device *dev, uint32_t channel)
 	return api->stop(dev, channel);
 }
 
+/**
+ * @brief request DMA channel.
+ *
+ * request DMA channel resources
+ * return -EINVAL if there is no valid channel available.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param filter_param filter function parameter
+ *
+ * @retval dma channel if successful.
+ * @retval Negative errno code if failure.
+ */
+__syscall int dma_request_channel(const struct device *dev,
+				  void *filter_param);
+
+static inline int z_impl_dma_request_channel(const struct device *dev,
```

Comment:
shouldn't this be in the `__subsystem struct dma_driver_api` as well?

Reply:
this is only used internal drivers no need to expose to use application.

Reply:
Ok, so the `__syscall` still makes sense?

Reply:
because other dirver may call this function. For example, ADC driver to have dma support it will call dma_request_channel.

Reply:
Ok, but if this call is not exposed to the application, does it still need to be exposed as a `__syscall`?

Reply:
the __syscall is
` 
#define __syscall static inline

if not staict inline this fucntion need move to .h file right?

Reply:
The point is that `__syscall` (and `z_impl_*`) is denoting something that can be called by an application running in userspace. Now, if this is not the case I don't see the need to have this function marked as `__syscall`

Reply:
ok, remove it.